src/tokens.js

```diff
@@ -303,34 +332,66 @@ const tokens = {
   /**
    * Generate mint data
    *
-   * @param {string} txID Hash of the transaction to be used to mint tokens
-   * @param {number} index Index of the output being spent
-   * @param {string} addressSpent Address of the output being spent
+   * @param {Object} mintInput tx containing mint authority {
+   *   {string} tx_id Hash of the tx
+   *   {number} index Index of the output being spent
+   *   {string} address The address associated with (tx_id, index)
+   * }
    * @param {string} token Token uid to be minted
    * @param {string} address Address to receive the amount of the generated token
    * @param {number} amount Amount of the token that will be minted
+   * @param {Object} depositInputs Inputs used for the HTR deposit (optional) {
+   *   {Object} inputs List of the inputs [{
+   *     {string} tx_id Hash of the tx
+   *     {number} index Index of the output
+   *     {string} address The address associated with (tx_id, index)
+   *     {number} token Token info, which should always be 0 in this case
+   *   }]
+   *   {number} amount Sum of the inputs. It might be more than the required deposit, so a change output will be created
+   * }
    * @param {Object} options {
    *   {boolean} createAnotherMint If should create another mint output after spending this one
    *   {boolean} createMelt If should create a melt output (useful when creating a new token)
    * }
    *
+   * @throws {InsufficientTokensError} If not enough tokens for deposit
+   *
    * @return {Object} Mint data {'inputs', 'outputs', 'tokens'}
    *
    * @memberof Tokens
    * @inner
    */
-  createMintData(txID, index, addressSpent, token, address, amount, options) {
+  createMintData(mintInput, token, address, amount, depositInputs, options) {
     const fnOptions = Object.assign({
       createAnotherMint: true,
       createMelt: false,
     }, options);
 
     const { createAnotherMint, createMelt } = fnOptions;
+    const inputs = [];
+    const outputs = [];
+
+    if (!depositInputs) {
+      // select HTR deposit inputs
+      const depositInfo = this.getMintDepositInfo(amount);
+      inputs.push(...depositInfo.inputs);
+      outputs.push(...depositInfo.outputs);
+    } else {
+      // user chose HTR deposit inputs
+      inputs.push(...depositInputs.inputs);
+      // create change output, if needed
+      const depositAmount = helpers.getDepositAmount(amount);
+      if (depositInputs.amount - depositAmount > 0) {
```

Comment:
If `depositInputs.amount - depositAmount < 0` you should raise an exception because the `mintInput` selected does not have sufficient amount for the deposit

Reply:
I agree.

Reply:
Done